torch/distributed/rpc/backend_registry.py

```diff
@@ -240,8 +238,16 @@ def _tensorpipe_exchange_and_check_all_device_maps(
         devices_set.discard(torch.device("cpu"))
         my_devices = list(devices_set)
     my_devices = sorted(my_devices, key=lambda d: d.index)
+    return my_devices
 
-    return reverse_device_maps, my_devices
+def _create_reverse_mapping(my_name, all_names, all_device_maps):
+    reverse_device_maps: Dict[str, Dict[torch.device, torch.device]] = {}
+    for node in all_names:
+        if my_name in all_device_maps[node]:
+            reverse_device_maps[node] = {
+                v: k for k, v in all_device_maps[node][my_name].items()
```

Comment:
need to verify this device is valid on this current node?

Reply:
Added validation logic for the device maps in func `_validate_device_maps()`